bundle/edu.gemini.spModel.smartgcal/src/main/java/edu/gemini/spModel/smartgcal/provider/CalibrationProviderImpl.java

```diff
@@ -33,18 +33,30 @@ public CalibrationProviderImpl(CalibrationRepository repository) {
 
     @Override
     public List<Calibration> getCalibrations(CalibrationKey key) {
-        ConfigurationKey configKey = key.getConfig();
+        final ConfigurationKey configKey = key.getConfig();
+        final List<Calibration> cals;
         if (key instanceof CalibrationKeyImpl.WithWavelength) {
-            Double wavelength = ((CalibrationKeyImpl.WithWavelength) key).getWavelength();
-            return getCalibrations(
-                    configKey.getInstrumentName(),
-                    configKey,
-                    wavelength);
+            final Double wavelength = ((CalibrationKeyImpl.WithWavelength) key).getWavelength();
```

Comment:
Good question.  I'm very hesitant to refactor Smart GCal though.  I just made cosmetic changes to these lines (added a `final` and removed the `return`s because I need to manipulate the result before handing it back.

The implementation in `ocs3` is straightforward, thanks to the new sequence model and the database, so I'll just leave further improvements until `ocs3`.